sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -388,3 +390,36 @@ object PhysicalWindow {
     case _ => None
   }
 }
+
+object ExtractSingleColumnNotInSubquery extends JoinSelectionHelper {
```

Comment:
This optimization is related only to sub-queries? Actually, this PR targets at null-aware anti-join with a single-column case? e.g., `select * from t1 anti join t2 on t1.id = t2.id or isnull(t1.id = t2.id)`

Reply:
this PR targets at null-aware anti-join with a single-column case.
this is a more specific version of description. I will update it. thanks

Reply:
Yea, thanks. Keeping the PR title/description updated is important for making other reviewers understood smoothly.